components/brave_new_tab_ui/components/default/rewards/index.tsx

```diff
@@ -0,0 +1,269 @@
+      return getLocale('rewardsWidgetTurningOn')
+    }
+
+    return getLocale('rewardsWidgetTurnOnRewards')
+  }
+
+  optInAction = (hasEnabled: boolean) => {
+    if (hasEnabled) {
+      this.props.onEnableRewards()
+    } else {
+      this.props.onCreateWallet()
+    }
+  }
+
+  renderPreOptIn = () => {
+    const {
+      enabledMain,
+      walletCreated
+    } = this.props
+
+    if (enabledMain && walletCreated) {
+      return null
+    }
+
+    const hasEnabled = !enabledMain && walletCreated
+
+    return (
+      <>
+        <PreOptInInfo>
+          <Title>
+            {
+              hasEnabled
+              ? getLocale('rewardsWidgetReEnableTitle')
+              : getLocale('rewardsWidgetEnableTitle')
+            }
+          </Title>
+          <SubTitle>
+            {
+              hasEnabled
+              ? getLocale('rewardsWidgetReEnableSubTitle')
+              : getLocale('rewardsWidgetEnableSubTitle')
+            }
+          </SubTitle>
+        </PreOptInInfo>
+        <PreOptInAction>
+          <TurnOnButton
+            data-test-id={'optInAction'}
+            onClick={this.optInAction.bind(this, hasEnabled)}
+          >
+            {this.getButtonText()}
+          </TurnOnButton>
+        </PreOptInAction>
+      </>
+    )
+  }
+
+  renderAmountItem = (type: AmountItemType) => {
+    const {
+      balance,
+      enabledAds,
+      onEnableAds,
+      adsEstimatedEarnings,
+      onlyAnonWallet,
+      totalContribution
+    } = this.props
+
+    const rates = balance.rates || {}
+    const showEnableAds = type === AmountItemType.ADS && !enabledAds
+    const amount = type === AmountItemType.TIPS
+      ? totalContribution
+      : adsEstimatedEarnings.toFixed(1)
+    const converted = convertBalance(amount, rates)
+    const batFormatString = onlyAnonWallet ? getLocale('rewardsWidgetBatPoints') : getLocale('rewardsWidgetBat')
+
+    return (
+      <AmountItem>
+        {
+          showEnableAds
+          ? <TurnOnAdsButton onClick={onEnableAds}>
+              {this.getButtonText(true)}
+            </TurnOnAdsButton>
+          : null
+        }
+        {
+          !showEnableAds
+          ? <AmountInformation data-test-id={`widget-amount-total-${type}`}>
+              <Amount>{amount}</Amount>
+              <ConvertedAmount>
+                {`${batFormatString} ${converted} USD`}
+              </ConvertedAmount>
+            </AmountInformation>
+          : null
+        }
+        <AmountDescription>
+          {
+            type === AmountItemType.ADS
+            ? getLocale('rewardsWidgetEstimatedEarnings')
+            : getLocale('rewardsWidgetMonthlyTips')
+          }
+        </AmountDescription>
+      </AmountItem>
+    )
+  }
+
+  renderRewardsInfo = () => {
+    const {
+      enabledMain,
+      walletCreated
+    } = this.props
+
+    if (!enabledMain || !walletCreated) {
+      return null
+    }
+
+    return (
+      <div data-test-id2={'enableMain'}>
```

Comment:
What's with the `data-test-id2` instead of simply `data-test-id`? If we ever want multiple values then we can space separate them and use the attribute word selector, e.g. `[data-test-id~="enableMain"]`.

Reply:
I just did this for parity with what's expected in the rewards_service browser tests:

https://github.com/brave/brave-core/blob/master/components/brave_rewards/browser/rewards_service_browsertest.cc#L649

Reply:
hmm ok well looks like we never need data-test-id**2** cc @NejcZdovc .

But for this, we don't have a browser test pointed at the NTP for rewards enabled, do we?

Reply:
I've made use of the previous test utility for testing enabling rewards and added a switch to test NTP:

https://github.com/brave/brave-core/pull/3624/files#diff-3a9e29d2367272c558efa9e9caf5a8f5